clickable.py

```diff
@@ -1214,17 +1213,122 @@ def _build(self):
         self.run_container_command(gocommand)
 
 
-class CordovaClickable(Clickable):
+class CordovaClickable(CMakeClickable):
+    # Lots of this code was based off of this:
+    # https://github.com/apache/cordova-ubuntu/blob/28cd3c1b53c1558baed4c66cb2deba597f05b3c6/bin/templates/project/cordova/lib/build.js#L59-L131
+    def __init__(self, *args, **kwargs):
+        super(CMakeClickable, self).__init__(*args, **kwargs)
+
+        self.platform_dir = os.path.join(self.cwd, 'platforms/ubuntu/')
+
+        self._dirs = {
+                'build'  : '{}/{}/{}/build/' .format(self.platform_dir, self.config.sdk, self.build_arch),
+                'prefix' : '{}/{}/{}/prefix/'.format(self.platform_dir, self.config.sdk, self.build_arch),
+                'make'   : '{}/build'.format(self.platform_dir)
+        }
+
+        self.temp = self._dirs['build']
+
+        self.config.specificDependencies = True
+        self.config.dependencies = [ # Got this list from check_reqs or somewhere
+                "libicu-dev:armhf",
+                "pkg-config",
+                "qtbase5-dev:armhf",
+                "qtchooser",
+                "qtdeclarative5-dev:armhf",
+                "qtfeedback5-dev:armhf",
+                "qtlocation5-dev:armhf",
+                "qtmultimedia5-dev:armhf",
+                "qtpim5-dev:armhf",
+                "libqt5sensors5-dev:armhf",
+                "qtsystems5-dev:armhf"
+        ]
+
+        if not os.path.isdir(self.platform_dir):
+            # fail when not using docker, need it anyways
+            if self.config.container_mode or self.config.lxd or self.config.chroot:
+                print_error('Docker is required to intialize cordova directories. Enable docker or run "cordova platform add ubuntu" manually to remove this message')
```

Comment:
Even if the user runs that cordova command this code will still be run. I think you can just drop that second sentence of the warning.

Reply:
running `cordova platform add ubuntu` would create the platform directory, and the conditional `os.pth.isdir(self.platform_dir)` would be true so the code wouldnt be run

Reply:
Ah, good point, missed that part!